lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -64,18 +65,33 @@ class DeployTask
     # core/v1/ReplicationController -- superseded by deployments/replicasets
     # extensions/v1beta1/ReplicaSet -- managed by deployments
     # core/v1/Secret -- should not committed / managed by shipit
-    PRUNE_WHITELIST = %w(
-      core/v1/ConfigMap
-      core/v1/Pod
-      core/v1/Service
-      batch/v1/Job
-      extensions/v1beta1/DaemonSet
-      extensions/v1beta1/Deployment
-      apps/v1beta1/Deployment
-      extensions/v1beta1/Ingress
-      apps/v1beta1/StatefulSet
-      autoscaling/v1/HorizontalPodAutoscaler
-    ).freeze
+    def prune_whitelist
+      wl = %w(
+        core/v1/ConfigMap
+        core/v1/Pod
+        core/v1/Service
+        batch/v1/Job
+        extensions/v1beta1/DaemonSet
+        extensions/v1beta1/Deployment
+        apps/v1beta1/Deployment
+        extensions/v1beta1/Ingress
+        apps/v1beta1/StatefulSet
+        autoscaling/v1/HorizontalPodAutoscaler
+      )
+      if server_version >= Gem::Version.new('1.8.0')
+        wl.concat(%w(
+          apps/v1beta2/DaemonSet
+          apps/v1beta2/Deployment
+          apps/v1beta2/StatefulSet
```

Comment:
These three shouldn't be necessary, since the group versions included in the original list still exist in 1.8.